### PR TITLE
Render precision save

### PR DESCRIPTION
**Pull request by DmytroMuravskyi**
### Purpose

"Render precision" option now remembered between sessions. Requested by Alias team. 
Also fixed typo in reference to ProtoGeometry

### Declarations

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers
@mjkkirschner @aparajit-pratap

### CODE REVIEW

**Comment by aparajit-pratap** [comment]
@DmytroMuravskyi looks good. Thanks for the changes!

**Comment by mjkkirschner** [comment]
Hi @DmytroMuravskyi can you add a test for this functionality - 
here is an example:
https://github.com/DynamoDS/Dynamo/blob/ec10f936824152e7dd7d6d019efdcda0d78a5264/test/DynamoCoreWpfTests/CoreUITests.cs#L432

Also what happens if a user's preference file does not have the render precision value serialized - does deserialization fail- does it default to some value?

**Comment by mjkkirschner** [comment]
``` "C:\projects\dynamo\src\Dynamo.All.sln" (default target) (1) ->
"C:\projects\dynamo\src\DynamoSandbox\DynamoSandbox.csproj.metaproj" (default target) (14) ->
"C:\projects\dynamo\src\DynamoSandbox\DynamoSandbox.csproj" (default target) (15) ->
"C:\projects\dynamo\src\DynamoCoreWpf\DynamoCoreWpf.csproj" (default target) (18) ->
(CoreCompile target) -> 
  ViewModels\RenderPackageFactoryViewModel.cs(49,132): error CS1061: 'IPreferences' does not contain a definition for 'RenderPrecision' and no extension method 'RenderPrecision' accepting a first argument of type 'IPreferences' could be found (are you missing a using directive or an assembly reference?) [C:\projects\dynamo\src\DynamoCoreWpf\DynamoCoreWpf.csproj]
    1465 Warning(s)
    1 Error(s)
Time Elapsed 00:00:34.73
 ```

File: test/DynamoCoreWpfTests/CoreUITests.cs [hunk 1]
```diff
@@ -468,6 +468,32 @@ public void PreferenceSetting_BackgroundPreview_1_0API()
             #endregion
         }
 
+        [Test, RequiresSTA]
+        [Category("DynamoUI")]
+        public void PreferenceSetting_RenderPrecision()
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/CoreUITests.cs]: could you test what happens if we encounter an old settings file that does not include the render precision property?

File: src/DynamoCoreWpf/ViewModels/RenderPackageFactoryViewModel.cs [hunk 2]
```diff
@@ -49,5 +50,14 @@ public RenderPackageFactoryViewModel(IPreferences preferenceSettings)
                 TessellationParameters = { ShowEdges = preferenceSettings.ShowEdges }
             };
         }
+
+        // TODO: Remove this constructor once the IPreferences and IRenderPrecisionPreferences have been consolidated in 3.0
+        public RenderPackageFactoryViewModel(PreferenceSettings preferenceSettings)
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/RenderPackageFactoryViewModel.cs]: I think actually the addition of this constructor may be a breaking change - this is an overload with the same parity as another - some calls will be ambiguous I think.

Instead what about checking the type with `is` in the other constructor?

File: src/DynamoCoreWpf/ViewModels/RenderPackageFactoryViewModel.cs [hunk 3]
```diff
@@ -44,10 +45,22 @@ public int MaxTessellationDivisions
...
+                this.factory = new HelixRenderPackageFactory()
+                {
+                    TessellationParameters = { ShowEdges = ps.ShowEdges, MaxTessellationDivisions = ps.RenderPrecision }
+                };
+            }
+            else
+            {
+                this.factory = new HelixRenderPackageFactory()
+                {
+                    TessellationParameters = { ShowEdges = preferenceSettings.ShowEdges}
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/RenderPackageFactoryViewModel.cs]: 👍

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 4]
```diff
@@ -324,6 +328,7 @@ public PreferenceSettings()
             UseHardwareAcceleration = true;
             PackageDownloadTouAccepted = false;
             maxNumRecentFiles = DefaultMaxNumRecentFiles;
+            RenderPrecision = 128;
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/PreferenceSettings.cs]: Can you define a const with comments for this magic number 😉

File: test/DynamoCoreWpfTests/CoreUITests.cs (hunk 1 above)
Comment (QilongTang) [reply test/DynamoCoreWpfTests/CoreUITests.cs]: Same

Comment (scottmitchell) [reply test/DynamoCoreWpfTests/CoreUITests.cs]: No problem

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 5]
```diff
@@ -27,6 +27,11 @@ public class PreferenceSettings : NotificationObject, IPreferences
         /// </summary>
         internal const int DefaultMaxNumRecentFiles = 10;
 
+        /// <summary>
+        /// Indicates the default render precision, i.e. the maximum number of tessellation divisions
+        /// </summary>
+        internal const int DefaultRenderPrecision = 128;
```
Comment (QilongTang) [inline src/DynamoCore/Configuration/PreferenceSettings.cs]: 👍

**Pull request merged**